src/misc/book-rewrite.ts

```diff
@@ -0,0 +1,28 @@
+import { Material } from 'org.bukkit';
+import { PlayerInteractEvent } from 'org.bukkit.event.player';
+import { EquipmentSlot, PlayerInventory } from 'org.bukkit.inventory';
+import { BookMeta } from 'org.bukkit.inventory.meta';
+
+// Players can re-open written books when they have a feather in offhand
+registerEvent(PlayerInteractEvent, (event) => {
+  if (event.hand !== EquipmentSlot.HAND) return;
+  if (event.item?.type !== Material.WRITTEN_BOOK) return;
+
+  // Check if player has feather
+  const inventory = event.player.inventory as PlayerInventory;
+  const offHand = inventory.itemInOffHand;
+  if (offHand.type !== Material.FEATHER) return;
+
+  const book = inventory.itemInMainHand;
+  const author = (book.itemMeta as BookMeta).author;
+  const username = event.player.name;
+  event.setCancelled(true);
+
+  if (author === username) {
+    book.type = Material.WRITABLE_BOOK; // Keeps the content of the book
+  } else {
+    event.player.sendActionBar(
```

Comment:
Why are we limiting this for only original author? To prevent book "griefing"?

Reply:
I think that the original author should be able to change his own books. If he wants others to be able to edit those, he can leave the book unsigned. Also, the original author limitation is defined in the Trello card, but we can change the limitation in the future if needed.